lib/connection.js

```diff
@@ -24,12 +24,53 @@ var EndpointState = require('./endpoint.js');
 var Session = require('./session.js');
 var Transport = require('./transport.js');
 
+var fs = require('fs');
+var os = require('os');
+var path = require('path');
 var net = require('net');
 var tls = require('tls');
 var EventEmitter = require('events').EventEmitter;
 
 var AMQP_PROTOCOL_ID = 0x00;
 
+function find_connect_config() {
+    var paths;
+    if (process.env.MESSAGING_CONNECT_FILE) {
```

Comment:
Just a thought, would exposing an option `connect_file_path` in the ConnectionOptions along with supporting the environment variable be a better idea?
Order of precedence
1.  connect_file_path in the options object
2. environment variable  `MESSAGING_CONNECT_FILE`

Reply:
I don't think it is needed. If the user wants to retrieve config from a file that is of course possible and easy to do in their application, and indeed will allow more options to be included. What is being added is a deliberately simple, minimal behaviour.